### ISSUE TITLE
error lnk2001

### ISSUE THREAD

**Issue by naccuracy** [zenkovich/o2#26]
Error	LNK2001	unresolved external symbol "private: static class o2::Type * o2::Actor::type" (?type@Actor@o2@@0PAVType@2@A)	o2Editor	o2\o2Editor\Platforms\Windows\ActorsTree.obj

всего 11 сообщений таких при сборке редактора, еще вопрос зачем нужна переменная О2? ее нельзя заменить относительными ссылками типа  ..\..\..\ ?

**Comment by naccuracy** [comment]
еще в описании C# изменено на lua, а в  разделе архитектура по прежнему написано про  C#.

**Comment by naccuracy** [comment]
Ничего не менял, собрал Code tool потом опять эдитор, получил диф! О_О
[my.diff.txt](https://github.com/zenkovich/o2/files/819292/my.diff.txt)
И самое главное что редактор собрался и запустился. что происходит?

**Comment by zenkovich** [comment]
Добавил файл кеша code tool в игнор, теперь дифа такого не должно быть

**Comment by naccuracy** [comment]
Ок, а закомитил? А то на свежем мастере у меня диф опять появился и кстати кроме кеша, еще диф в коде и заголовочных файлах: в некоторых местах просто +- пустая строка - это наверное из-за того, что я использую кнсольный гит, а в некоторых файлах добавлены строки кода, так что с этим нужно разобраться. Еще я бы сделал отдельную папку для релизных бинарей, и туда ручками или скриптом складывал стабильные версии, а тот, что генерится студией, в результате компиляции в гит игнор занести было бы неплохо.

**Comment by naccuracy** [comment]
+- пустая строка - результат работы CodeTool, что то с вставкой кода - с символами \0 и \n

**Comment by zenkovich** [comment]
Да, буквально пол часа назад залил версию с почти работающими прототипами, там вроде появилась мета-информация для актора.
Насчет пустых строк я пока не парюсь. А насчет бинарей - да, в будущем хочу делать коммиты только с работающим бинарем. Да и сейчас стараюсь при каждом коммите обновлять релизную сборку, чтобы в ней были актуальные изменения. Но это все ни разу не стабильно, багов куча, времени их все править нет, да и смысла особо тоже нет, потому что все меняется и взаимоломается.

**Comment by naccuracy** [comment]
убрал нулевые байты из всех файлов, вроде, заигнорил и удалил кеш, подправил запись файлов в codetool, запушил пересобрал и в диф попал только экзешник. зацени последние 2 комита, ничего я не поломал?
